<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1022</width>
    <height>960</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>hc_monitor</string>
  </property>
  <property name="animated">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QWidget" name="">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0,0,0">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QLabel" name="host_label">
            <property name="text">
             <string>Host:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="host"/>
          </item>
          <item>
           <widget class="QLabel" name="port_label">
            <property name="text">
             <string>Port:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="port">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="connect">
            <property name="text">
             <string>&amp;Connect</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="handleWidth">
           <number>4</number>
          </property>
          <widget class="QWidget" name="">
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label_rows">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Rows</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="probe_rows">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>99999999</number>
              </property>
              <property name="value">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_cols">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Columns</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="probe_cols">
              <property name="maximum">
               <number>9999999</number>
              </property>
              <property name="value">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_width">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Width</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QDoubleSpinBox" name="probe_area_width">
              <property name="maximum">
               <double>9999999.000000000000000</double>
              </property>
              <property name="value">
               <double>30.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="QLabel" name="label_height">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Height</string>
              </property>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="QDoubleSpinBox" name="probe_area_height">
              <property name="maximum">
               <double>9999999.000000000000000</double>
              </property>
              <property name="value">
               <double>30.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="8" column="0">
             <widget class="QLabel" name="margin">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Margin</string>
              </property>
             </widget>
            </item>
            <item row="8" column="1">
             <widget class="QDoubleSpinBox" name="probe_margin">
              <property name="minimum">
               <double>-99999.000000000000000</double>
              </property>
              <property name="maximum">
               <double>99999.990000000005239</double>
              </property>
              <property name="value">
               <double>-1.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="10" column="0">
             <widget class="QLabel" name="lalel_start_z">
              <property name="text">
               <string>Start Z</string>
              </property>
             </widget>
            </item>
            <item row="10" column="1">
             <widget class="QSpinBox" name="probe_start_z">
              <property name="value">
               <number>0</number>
              </property>
             </widget>
            </item>
            <item row="11" column="1">
             <widget class="QPushButton" name="run_probe">
              <property name="text">
               <string>Run probe</string>
              </property>
             </widget>
            </item>
            <item row="12" column="1">
             <widget class="QPushButton" name="process">
              <property name="text">
               <string>Process</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="">
           <layout class="QVBoxLayout" name="verticalLayout">
            <property name="spacing">
             <number>6</number>
            </property>
            <item>
             <widget class="QTextEdit" name="text">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>50</height>
               </size>
              </property>
              <property name="baseSize">
               <size>
                <width>0</width>
                <height>50</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTreeWidget" name="comtree">
              <property name="verticalScrollMode">
               <enum>QAbstractItemView::ScrollPerPixel</enum>
              </property>
              <property name="columnCount">
               <number>3</number>
              </property>
              <column>
               <property name="text">
                <string notr="true">1</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string notr="true">2</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string notr="true">3</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>Command:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="prompt">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="editable">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="send">
                <property name="text">
                 <string>&amp;Send</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="HCViewerWidget" name="gl" native="true">
       <property name="minimumSize">
        <size>
         <width>500</width>
         <height>500</height>
        </size>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1022</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="actionQuit"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_G_code"/>
    <addaction name="actionSave_G_code"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_probe_data"/>
    <addaction name="actionSave_probe_data"/>
    <addaction name="separator"/>
    <addaction name="actionSave_probe_G_code"/>
    <addaction name="separator"/>
    <addaction name="actionSave_log"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionLoad_probe_data">
   <property name="text">
    <string>&amp;Load probe data</string>
   </property>
  </action>
  <action name="actionSave_probe_data">
   <property name="text">
    <string>Save &amp;probe data</string>
   </property>
  </action>
  <action name="actionLoad_G_code">
   <property name="text">
    <string>Load &amp;G-code</string>
   </property>
  </action>
  <action name="actionSave_G_code">
   <property name="text">
    <string>Save G-&amp;code</string>
   </property>
  </action>
  <action name="actionSave_probe_G_code">
   <property name="text">
    <string>Sa&amp;ve probe G-code</string>
   </property>
  </action>
  <action name="actionSave_log">
   <property name="text">
    <string>Sav&amp;e log</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>HCViewerWidget</class>
   <extends>QWidget</extends>
   <header>viewerwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
